src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -1940,8 +1940,15 @@ class MyClass
                 ' Get cached project diagnostics.
                 Dim diagnostics = diagnosticService.GetCachedDiagnosticsAsync(workspace, project.Id).WaitAndGetResult(CancellationToken.None)
 
-                Assert.Equal(1, diagnostics.Count())
-                Assert.Equal(HiddenDiagnosticsCompilationAnalyzer.Descriptor.Id, diagnostics.Single().Id)
+                ' different behavior between v1 and v2. in v2. in v2, solution crawler never creates non-local hidden diagnostics.
```

Comment:
adding @srivatsn 

I think we should cut this until we have option page where user can choose what analyzers on what scope.

the reason we don't run hidden analyzers on close file was perf and then right after that turning it on for whole compilation is same thing as what we did for close file diagnostic options.

if the decision was for perf, we should not switch the stand in the middle of it for functionality.

we should enable it when we can support it which I think adding option page where user can see perf of each analyzer and can set when those runs in what orders.